openstack/datadog_checks/openstack/openstack.py

```diff
@@ -1015,32 +1024,26 @@ def _is_valid_metric(label):
 
         project_name = project.get('name')
 
-        self.log.debug("Collecting metrics for project. name: {0} id: {1}".format(project_name, project['id']))
+        self.log.debug("Collecting metrics for project. name: {} id: {}".format(project_name, project['id']))
 
-        url = '{0}/limits'.format(self.get_nova_endpoint())
+        url = '{}/limits'.format(self.get_nova_endpoint())
         headers = {'X-Auth-Token': self.get_auth_token()}
         server_stats = self._make_request_with_auth_fallback(url, headers, params={"tenant_id": project['id']})
 
-        server_tags.append('tenant_id:{0}'.format(project['id']))
+        server_tags.append('tenant_id:{}'.format(project['id']))
 
         if project_name:
-            server_tags.append('project_name:{0}'.format(project['name']))
+            server_tags.append('project_name:{}'.format(project['name']))
 
         for st in server_stats['limits']['absolute']:
```

Comment:
This could raise a KetError if `server_stats` is empty. I am not sure if this is valid.

Reply:
I think this is relatively safe since the only time this should not have this section is if the request returned an exception, which would bubble up and cause a backoff. What do you think?

Reply:
I am not sure how this code will behave with other nova versions and such. In general, I like it better when we rely on well defined and localized exceptions (where we can) rather than doing a catch all the things strategy (which is still needed anyway)